airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/source.py

```diff
@@ -523,19 +550,11 @@ def check_connection(self, logger: logging.Logger, config: Mapping[str, Any]) ->
                 # explicitly setting small page size for the check operation not to cause OOM issues
                 stream = GoogleAnalyticsDataApiMetadataStream(config=_config, authenticator=_config["authenticator"])
                 metadata = next(stream.read_records(sync_mode=SyncMode.full_refresh), None)
-            except HTTPError as e:
-                error_list = [HTTPStatus.BAD_REQUEST, HTTPStatus.FORBIDDEN]
-                if e.response.status_code in error_list:
-                    internal_message = f"Incorrect Property ID: {property_id}"
-                    property_id_docs_url = (
-                        "https://developers.google.com/analytics/devguides/reporting/data/v1/property-id#what_is_my_property_id"
-                    )
-                    message = f"Access was denied to the property ID entered. Check your access to the Property ID or use Google Analytics {property_id_docs_url} to find your Property ID."
-
-                    wrong_property_id_error = AirbyteTracedException(
-                        message=message, internal_message=internal_message, failure_type=FailureType.config_error
-                    )
-                    raise wrong_property_id_error
+            except (MessageRepresentationAirbyteTracedErrors, BaseBackoffException) as ex:
+                if hasattr(ex, "failure_type") and ex.failure_type == FailureType.config_error:
+                    # bad request and forbidden are set in mapper as config errors
+                    raise ex
+                logger.error(f"Check failed", exc_info=ex)
```

Comment:
e.- 
_get_error_mapping_ is implemented in **GoogleAnalyticsDataApiMetadataStream**. 

Previously, _check_connection_ in **SourceGoogleAnalyticsDataApi** would format and raise an "Incorrect Property ID:..." message for 400, 403 HTTP exceptions checking the connection (read records). Now the method _get_google_analytics_data_api_metadata_error_mapping_ handles this task.